pylhc/sixdesk_tools/create_workspace.py

```diff
@@ -0,0 +1,192 @@
+    _write_mask(jobname, basedir, mask_text, **kwargs)
+
+    start_subprocess([SETENV_SH, "-s"], cwd=sixjobs_path, ssh=ssh)
+    LOG.info("Workspace fully set up.")
+
+
+def remove_twiss_fail_check(jobname: str, basedir: Path):
+    """ Comments out the "Twiss fail" check from mad6t.sh """
+    LOG.info("Applying twiss-fail hack.")
+    for mad6t_path in (
+        get_mad6t_mask_path(jobname, basedir),
+        get_mad6t1_mask_path(jobname, basedir),
+    ):
+        with open(mad6t_path, "r") as f:
+            lines = f.readlines()
+
+        check_started = False
+        for idx, line in enumerate(lines):
+            if line.startswith('grep -i "TWISS fail"'):
+                check_started = True
+
+            if check_started:
+                lines[idx] = f"# {line}"
+                if line.startswith("fi"):
+                    break
+        else:
+            LOG.info(f"'TWISS fail' not found in {mad6t_path.name}")
+            continue
+
+        with open(mad6t_path, "w") as f:
+            f.writelines(lines)
+
+
+# Helper -----------------------------------------------------------------------
+
+
+def _create_workspace(jobname: str, basedir: Path, ssh: str = None):
+    """ Create workspace structure (with default files). """
+    workspace_path = get_workspace_path(jobname, basedir)
+    scratch_path = get_scratch_path(basedir)
+    LOG.info(f'Creating new workspace in "{str(workspace_path)}"')
+
+    if workspace_path.exists():
+        LOG.warning(f'Workspace in "{str(workspace_path)}" already exists. ')
+        LOG.info("Do you want to delete the old workspace? [y/N]")
+        user_answer = input()
+        if user_answer.lower().startswith("y"):
+            shutil.rmtree(workspace_path)
+            try:
+                shutil.rmtree(scratch_path)
+            except FileNotFoundError:
+                pass
+        else:
+            LOG.warning("Keeping Workspace as-is.")
+            return
+
+    scratch_path.mkdir(parents=True, exist_ok=True)
+
+    # create environment with all necessary files
+    # _start_subprocess(['git', 'clone', GIT_REPO, basedir])
+    start_subprocess([SETENV_SH, "-N", workspace_path.name], cwd=basedir, ssh=ssh)
+
+    # create autosix results folder.
+    # Needs to be done after above command (as it crashes if folder exists)
+    # but before end of this stage (as it needs to write the stagefile)
+    get_autosix_results_path(jobname, basedir).mkdir(exist_ok=True, parents=True)
+
+
+def _create_sixdeskenv(jobname: str, basedir: Path, **kwargs):
+    """ Fills sixdeskenv mask and copies it to workspace """
+    workspace_path = get_workspace_path(jobname, basedir)
+    scratch_path = get_scratch_path(basedir)
+    sixdeskenv_path = get_sixdeskenv_path(jobname, basedir)
+
+    missing = [key for key in SIXENV_REQUIRED if key not in kwargs.keys()]
+    if len(missing):
+        raise ValueError(f"The following keys are required but missing {missing}.")
```

Comment:
Mentions "following keys required` but does not show them

Reply:
? the `{missing}` part?

Reply:
I'm retarded today, my bad